open_rarity/models/collection.py

```diff
@@ -120,6 +113,35 @@ def _normalize_attributes_frequency_counts(
     def token_total_supply(self) -> int:
         return len(self.tokens)
 
+    @cached_property
+    def has_numeric_attribute(self) -> bool:
+        return (
+            next(
+                filter(
+                    lambda t: len(t.metadata.numeric_attributes)
+                    or len(t.metadata.date_attributes),
+                    self.tokens,
+                ),
+                None,
+            )
+            is not None
+        )
+
+    @cached_property
+    def token_standards(self) -> list[TokenStandard]:
+        """Returns token standards for this collection.
+
+        Returns
+        -------
+        list[TokenStandard]
+            the set of unique token standards that any token in this collection
+            interfaces or uses.
+        """
+        token_standards = set()
+        for token in self.tokens:
```

Comment:
This would change to `self._tokens` if the aforementioned suggestion is adopted.